### DISCUSSION TITLE
Failed to download Resources

### DISCUSSION THREAD

**Discussion by psumanth254** [Xceptance/XLT#389] [General]
Hi team , 
i am trying to load one Url where its giving responce as failed to download resources.

Thanks

**Comment by psumanth254** [comment]
The below issue we are getting when we used LoadPage action used options like sending only URLString , URlString along with time , URL options for all the options we are getting same issue 
![MicrosoftTeams-image (4)](https://github.com/Xceptance/XLT/assets/78797315/6d87dd9c-7159-49da-9d5c-e56a5dd0f246)

**Comment by jowerner** [comment]
Can you please post the remaining part of the stack trace?

Which version of Java are you using to run your test case? If possible, use Java 11. Alternatively, edit file "config/default.properties" and switch socket instrumentation off:

```
com.xceptance.xlt.socket.collectNetworkData = false
```

**Reply by bhakarswayamavarapu** [reply]
Hi @jowerner, we are able to run the scripts properly after using java version 11. But in the virtual machines(provided by client), the java version 17 was installed. 
Can you please suggest what we have to follow here, whether the java version is to be updated to 17 in my local or can we ask the client to install java version 11 in the machines?
Thanks!

**Reply by jowerner** [reply]
Those virtual machines, are those the ones you will use as load test agent machines? If so, just give them a try even if they have Java 17 already. The XLT load agent is configured with the needed JDK command line options so socket instrumentation should work even with JDK 17.

For developing and testing your test scenarios from within your IDE on your local machine, we recommend to stay with Java 11. If you are forced to upgrade to Java 17, you will have to disable socket instrumentation in the test suite settings, preferably in file `<testsuite>/config/dev.properties`:

```
com.xceptance.xlt.socket.collectNetworkData = false
```

This way, when running test scenarios from your IDE, socket instrumentation is off (at this stage, you will typically not need that data anyway), while later when running the load test, socket instrumentation is on.

**Reply by bhakarswayamavarapu** [reply]
Hi @jowerner 
Thanks for the inputs.
So you mean to say that this will work even we have Java 17 installed in the virtual machine agents and Java 11 installed in my local machine?

**Reply by jowerner** [reply]
If you use those virtual machines as load test agents (i.e. running an XLT Agent Controller), which execute your test suite as a load test, then you should be good to go. Having Java 11 on your local machine and maybe even running the Master Controller for your load test with Java 11, is not an issue. But as always, trying it out beforehand is a good idea.

**Reply by bhakarswayamavarapu** [reply]
Sure thanks!

**Comment by psumanth254** [comment]
HI @jowerner , At present we are using java version "20.0.1" 2023-04-18 and we are not able to find the switch socket instrumentation off in default.properties 
if we add com.xceptance.xlt.socket.collectNetworkData = false We are not able to run tVisit script using 
Junit

**Reply by jowerner** [reply]
So what error do you get this time?

**Reply by psumanth254** [reply]
At present we are trying to update java version to 11 (from 20.0.1)
